addons/stock/models/stock_quant.py

```diff
@@ -45,13 +76,75 @@ class StockQuant(models.Model):
         'Reserved Quantity',
         default=0.0,
         help='Quantity of reserved products in this quant, in the default unit of measure of the product',
-        readonly=True, required=True)
+        readonly=True)
     in_date = fields.Datetime('Incoming Date', readonly=True)
+    tracking = fields.Selection(related='product_id.tracking', readonly=True)
+
+    def _set_inventory_quantity(self):
+        for quant in self:
+            # Get the quantity to create a move for.
+            rounding = quant.product_id.uom_id.rounding
+            diff = float_round(quant.inventory_quantity - quant.quantity, precision_rounding=rounding)
+            diff_float_compared = float_compare(diff, 0, precision_rounding=rounding)
+            # Create and vaidate a move so that the quant matches its `inventory_quantity`.
+            if diff_float_compared == 0:
+                continue
+            elif diff_float_compared > 0:
+                move_vals = self._get_inventory_move_values(diff, self.product_id.property_stock_inventory, self.location_id)
+            else:
+                move_vals = self._get_inventory_move_values(-diff, self.location_id, self.product_id.property_stock_inventory, out=True)
+            move = self.env['stock.move'].with_context(inventory_mode=False).create(move_vals)
+            move._action_done()
+
+    @api.model
+    def create(self, vals):
+        """ Override to handle the "inventory mode" and create a quant as superuser the conditions
+        are met.
+        """
+        if self._is_inventory_mode() and 'inventory_quantity' in vals:
+            allowed_fields = self._get_inventory_fields_create()
+            if any([field for field in vals.keys() if field not in allowed_fields]):
+                raise UserError(_("Quant's creation is restricted, you can't do this operation."))
+            inventory_quantity = vals.pop('inventory_quantity')
+
+            # Create an empty quant or write on a similar one.
+            product = self.env['product.product'].browse(vals['product_id'])
+            location = self.env['stock.location'].browse(vals['location_id'])
+            lot_id = self.env['stock.production.lot'].browse(vals.get('lot_id'))
+            package_id = self.env['stock.quant.package'].browse(vals.get('package_id'))
+            owner_id = self.env['res.partner'].browse(vals.get('owner_id'))
+            quant = self._gather(product, location, lot_id=lot_id, package_id=package_id, owner_id=owner_id, strict=True)
+            if not quant:
+                quant = self.sudo().create(vals)
+            # Set the `inventory_quantity` field to create the necessary move.
+            quant.inventory_quantity = inventory_quantity
+            return quant
+        return super(StockQuant, self).create(vals)
+
+    def read(self, fields=None, load='_classic_read'):
+        res = super(StockQuant, self).read(fields=fields, load=load)
+        for quant in res:
```

Comment:
""" Override to handle the "inventory mode" and fill in the `inventory_quantity` field at read. """
if self._is_inventory_mode() and 'inventory_quantity' in fields and 'quantity' in fields:
for quant in res:
    quant['inventory_quantity'] = quant['quantity']